test/unit/solver/test_cuda_rosenbrock.cpp

```diff
@@ -119,70 +143,71 @@ RosenbrockPolicy getSolver(std::size_t number_of_grid_cells)
       micm::RosenbrockSolverParameters::three_stage_rosenbrock_parameters(number_of_grid_cells, false));
 }
 
-template<template<class> class MatrixPolicy, template<class> class SparseMatrixPolicy, class LinearSolverPolicy>
+template<template<class> class CPUMatrixPolicy, template<class> class CPUSparseMatrixPolicy, class CPULinearSolverPolicy
+         template<class> class GPUMatrixPolicy, template<class> class GPUSparseMatrixPolicy, class GPULinearSolverPolicy>
 void testAlphaMinusJacobian(std::size_t number_of_grid_cells)
 {
   auto gpu_solver = getSolver<
-      MatrixPolicy,
-      SparseMatrixPolicy,
-      LinearSolverPolicy,
-      micm::CudaRosenbrockSolver<MatrixPolicy, SparseMatrixPolicy, LinearSolverPolicy>>(number_of_grid_cells);
-  auto jacobian = gpu_solver.GetState().jacobian_;
-  EXPECT_EQ(jacobian.size(), number_of_grid_cells);
-  EXPECT_EQ(jacobian[0].size(), 5);
-  EXPECT_EQ(jacobian[0][0].size(), 5);
-  EXPECT_GE(jacobian.AsVector().size(), 13 * number_of_grid_cells);
-  for (auto& elem : jacobian.AsVector())
+      GPUMatrixPolicy,
+      GPUSparseMatrixPolicy,
+      GPULinearSolverPolicy,
+      micm::CudaRosenbrockSolver<GPUMatrixPolicy, GPUSparseMatrixPolicy, GPULinearSolverPolicy>>(number_of_grid_cells);
+  auto gpu_jacobian = gpu_solver.GetState().jacobian_;
+  EXPECT_EQ(gpu_jacobian.size(), number_of_grid_cells);
+  EXPECT_EQ(gpu_jacobian[0].size(), 5);
+  EXPECT_EQ(gpu_jacobian[0][0].size(), 5);
+  EXPECT_GE(gpu_jacobian.AsVector().size(), 13 * number_of_grid_cells);
+  for (auto& elem : gpu_jacobian.AsVector())
     elem = 100.0;
   for (std::size_t i_cell = 0; i_cell < number_of_grid_cells; ++i_cell)
   {
-    jacobian[i_cell][0][0] = 12.2;
-    jacobian[i_cell][0][1] = 24.3 * (i_cell + 2);
-    jacobian[i_cell][0][2] = 42.3;
-    jacobian[i_cell][1][0] = 0.43;
-    jacobian[i_cell][1][1] = 23.4;
-    jacobian[i_cell][1][2] = 83.4 / (i_cell + 3);
-    jacobian[i_cell][2][0] = 4.74;
-    jacobian[i_cell][2][2] = 6.91;
-    jacobian[i_cell][3][1] = 59.1;
-    jacobian[i_cell][3][3] = 83.4;
-    jacobian[i_cell][4][0] = 78.5;
-    jacobian[i_cell][4][2] = 53.6;
-    jacobian[i_cell][4][4] = 1.0;
+    gpu_jacobian[i_cell][0][0] = 12.2;
+    gpu_jacobian[i_cell][0][1] = 24.3 * (i_cell + 2);
+    gpu_jacobian[i_cell][0][2] = 42.3;
+    gpu_jacobian[i_cell][1][0] = 0.43;
+    gpu_jacobian[i_cell][1][1] = 23.4;
+    gpu_jacobian[i_cell][1][2] = 83.4 / (i_cell + 3);
+    gpu_jacobian[i_cell][2][0] = 4.74;
+    gpu_jacobian[i_cell][2][2] = 6.91;
+    gpu_jacobian[i_cell][3][1] = 59.1;
+    gpu_jacobian[i_cell][3][3] = 83.4;
+    gpu_jacobian[i_cell][4][0] = 78.5;
+    gpu_jacobian[i_cell][4][2] = 53.6;
+    gpu_jacobian[i_cell][4][4] = 1.0;
   }
 
   // Negate the Jacobian matrix (-J) here
-  for (auto& elem : jacobian.AsVector())
+  for (auto& elem : gpu_jacobian.AsVector())
     elem = -elem;
```

Comment:
```suggestion
  std::transform(gpu_jacobian_vec.cbegin(), gpu_jacobian_vec.cend(), gpu_jacobian_vec.begin(), std::negate<>{});
```
This one might be a little overkill so if the original for loop is more readable, please disregard!

Reply:
Thanks. Done.